### ISSUE TITLE
No output for merge resolution?

### ISSUE THREAD

**Issue by dabrahams** [paulaltin/git-subline-merge#13]
I just had a merge result that surprised me; there was nothing in the output of git merge but the newlines that git-subline-merge seems to dump, but it looked something like this:

Ancestor:
```swift
                  let z = someFunc(
                      max(a / b, 0), in: c, by: d)

                 otherFunc(
                      &q[m], arg: n)
```

Branch 1:
```swift
                  let z = someFunc(
                      max(a / b, 0), in: c, by: d)

                 print("something")
                 otherFunc(
                      &q[m], arg: n)
```

Branch2:
```swift
                  let z = someFunc(
                      max(a / b, 0), in: c, by: d)

                 let b = otherFunc(
                      &q[m], arg: n)
```

Merge result:
```swift
                  let z = someFunc(
                      max(a / b, 0), in: c, by: d)

                 let b = print("something")
                 otherFunc(
                      &q[m], arg: n)
```

I think I understand why it got the result it did (though maybe there's a simple heuristic that better associates changes on the same line with one another?), but it surprised me that I didn't see a report of the resolution from git-subline-merge in my magit process buffer.  Huh, just tried from the command line and I see the report.  So this is either a git-subline-merge issue having to do with how it responds to being run under magit (e.g. response to TERM settings or something), or it's an issue with magit itself hiding some of the output.

**Comment by dabrahams** [comment]
Investigating: https://gitter.im/magit/magit?at=5db3370e9825bd6bacc9dc92

**Comment by dabrahams** [comment]
The full reproducer is [here](https://github.com/dabrahams/magit-subline-merge-bug)

**Comment by paulaltin** [comment]
Hi @dabrahams, thanks for reporting this. I'm following the conversation on Gitter, and trying to get Magit installed so I can test it too.

With regard to the merge result itself, yes I've seen some odd things like that before. I've actually been meaning to look into git's "patience" merging algorithm to see whether it performs better in these situations: your code looks like a nice test case for that.

Also, you mentioned the that GSM seems to dump newlines in the output – is this the same issue from #5? I thought I had fixed those, but perhaps not...

**Comment by dabrahams** [comment]
Unfortunately I can't post the actual code and when I tried to reproduce it in the small, it doesn't happen.

It appears that I was on the "print" branch and not on "master," could that account for the newline thing?

**Comment by dabrahams** [comment]
Looks like there are no changes between those two, so maybe not

**Comment by paulaltin** [comment]
The print branch was merged into master at bcc974dc (and I can't see it on github any more at all). Are you using the latest master version? I don't see any blank lines in the output in non-interactive mode, for example:

    First, rewinding head to replay your work on top of it...
    Applying: ex1
    Using index info to reconstruct a base tree...
    M	example.cpp
    Falling back to patching base and 3-way merge...
    git-subline-merge resolved conflict 1 of 2 in example.cpp, resulting hunk was:
    <<<<<<<
    // Cueball 1: Sudo make me a sandwich.
    // Cueball 2: Okay.
    >>>>>>>
    git-subline-merge resolved conflict 2 of 2 in example.cpp, resulting hunk was:
    <<<<<<<
    int myFunction(const Object& obj, const std::string_view& name) {
        obj.doSomething(name);
    >>>>>>>
    Resolved 2 of 2 conflicts in example.cpp
    Auto-merging example.cpp
